rust/src/router/pipe_transport.rs

```diff
@@ -159,10 +160,7 @@ impl PipeTransportDump {
                 .sctp_parameters
                 .as_ref()
                 .map(|parameters| SctpParameters::from_fbs(parameters.as_ref())),
-            sctp_state: dump
-                .base
-                .sctp_state
-                .map(|state| SctpState::from_fbs(&state)),
+            sctp_state: dump.base.sctp_state.map(SctpState::from_fbs),
```

Comment:
BTW I think you may also be able implement additional helper impls this (pseudo-code, may not compile):
```rust
impl<Source, Target> FromFbs for Option<Target>
where
    Source: FromFbs,
{
    type FbsType = Option<Source>;

    fn from_fbs(value: Self::FbsType) -> Self {
       value.map(Self::FbsType::from_fbs)
    }
}
```

Then also helper for inverse conversion like `From`/`Into` symmetry and many cases could be simplified:
```suggestion
            sctp_state: dump.base.sctp_state.to_fbs_source(),
```

Reply:
This implementation would take an `Option` of a type implementing `FromFbs`:

```rust
impl<T> FromFbs for Option<T>
where
    T: FromFbs,
{
    type FbsType = Option<T::FbsType>;

    fn from_fbs(value: Self::FbsType) -> Self {
        value.map(T::from_fbs)
    }
}
```

This can later be used as:

```rust
sctp_state: FromFbs::from_fbs(dump.base.sctp_state) // dump.base.sctp_state is of type Option<fbs::SctpState>
```

Reply:
Yes

Reply:
I've already done this. Since there are still `mediasoup` (no `mediasoup-types`) types that do have their own `to_fbs` and `from_fbs` rather than implementing the new traits, there is place for enhancement but I'm not going to implement all those traits in this PR.